encord/objects/ontology_object_instance.py

```diff
@@ -465,8 +478,47 @@ def set_for_frames(
             reviews: Should only be set by internal functions.
             is_deleted: Should only be set by internal functions.
         """
-        if self._range_only:
-            if not isinstance(coordinates, AudioCoordinates):
+        if range_html is not None:
+            if self._ontology_object.shape != Shape.TEXT:
+                raise LabelRowError(
+                    "Setting range_html of the object instance is only allowed for objects with the 'text' shape"
+                )
+            elif self._parent is not None and self._parent.file_type != "text/html":
+                raise LabelRowError("Cannot add range_html to a non-html text file")
+            elif len(self.range_list) > 0:
+                raise LabelRowError("Cannot add range_html to an object instance that has a non-html range")
```

Comment:
An important set of checks to ensure range_html is used properly on an ObjectInstance.